### ISSUE TITLE
What about the compatibility with flutter 2?

### ISSUE THREAD

**Issue by KevinHua** [waseemdev/vscode-flutter.xml-layout#11]
Thanks!

**Comment by abulka** [comment]
Agreed - is this flutter xml layout extension actually usable in Flutter 2, because I'm getting
```
Error: Cannot run with sound null safety, because the following dependencies
don't support null safety:

- package:flutter_xmllayout_helpers
```

and

```dart
import 'package:flutter_xmllayout_helpers/flutter_xmllayout_helpers.dart';
```

    The library 'package:flutter_xmllayout_helpers/flutter_xmllayout_helpers.dart' is legacy, 
    and should not be imported into a null safe library.
    Try migrating the imported library.dart(import_of_legacy_library_into_null_safe)
    flutter_xmllayout_helpers

And even when I use this launch config to turn off null safety:

```json
{
    "name": "xml_layout1 (debug mode)",
    "request": "launch",
    "type": "dart",
    "args": ["--no-sound-null-safety"],
    "flutterMode": "debug"
}
```

I still get

    ../../../../../andy/.pub-cache/hosted/pub.dartlang.org/flutter_xmllayout_helpers-0.0.9/lib/forms/FormGroup.dart:110:34: Error: The getter 'Observable' isn't defined for the class 'FormGroup'.
    - 'FormGroup' is from 'package:flutter_xmllayout_helpers/forms/FormGroup.dart' ('../../../../../andy/.pub-cache/hosted/pub.dartlang.org/flutter_xmllayout_helpers-0.0.9/lib/forms/FormGroup.dart').
    Try correcting the name to the name of an existing getter, or defining a getter or field named 'Observable'.
        _submitEnabledSubscription = Observable.merge([_statusStream, _submitting]).listen((data) {

                                    ^^^^^^^^^^
    Failed to compile application.

This looks like a great tool - any chance of an update?

**Comment by abulka** [comment]
I ditched trying to use the old `provider` and the problems it brings with the latest Flutter 2.  Luckily it looks like `flutter_xmllayout_helpers` can be updated to use the latest `provider: ^5.0.0` 

> RxDart 0.23.x moves away from the Observable class, utilizing Dart 2.6's new extension methods instead. This requires several small refactors that can be easily automated -- which is just what we've done!  Please follow the instructions on the [rxdart_codemod](https://pub.dev/packages/rxdart_codemod) package to automatically upgrade your code to support RxDart 0.23.x.

Running the above tool made the following changes which just seem to be replacing `Observable` with `Rx`:

```diff
+++ b/lib/forms/FormGroup.dart
@@ -80,8 +80,7 @@ class FormGroup {
       _controlsStatusSubscription.cancel();
       _controlsStatusSubscription = null;
     }
-    _controlsStatusSubscription = Observable
-      .merge(streams)
+    _controlsStatusSubscription = Rx.merge(streams)
       .listen((value) {
         final old = _statusStream.value;
         if (value != old) {
@@ -107,7 +106,7 @@ class FormGroup {
       _submitEnabledSubscription.cancel();
       _submitEnabledSubscription = null;
     }
-    _submitEnabledSubscription = Observable.merge([_statusStream, _submitting]).listen((data) {
+    _submitEnabledSubscription = Rx.merge([_statusStream, _submitting]).listen((data) {
```

**Comment by Jamesweng** [comment]
We're using this extension with Flutter 2 in our project, with null-safety disabled and following package overrides the extension run well - 

dependency_overrides:
  path_provider: ^2.0.1
  rxdart: ^0.22.0
  xml: ^5.0.0
